go/libraries/doltcore/env/multi_repo_env.go

```diff
@@ -364,13 +364,47 @@ func MultiEnvForPaths(
 		} else if dEnv.CfgLoadErr != nil {
 			return nil, fmt.Errorf("error loading environment '%s' at path '%s': %s", name, absPath, dEnv.CfgLoadErr.Error())
 		}
+		envSet[name] = dEnv
+	}
 
-		mrEnv.AddEnv(name, dEnv)
+	enforceSingleFormat(envSet)
+	for dbName, dEnv := range envSet {
+		mrEnv.AddEnv(dbName, dEnv)
 	}
 
 	return mrEnv, nil
 }
 
+// enforceSingleFormat enforces that constraint that all databases in
+// a multi-database environment have the same NomsBinFormat.
+// Databases are removed from the MultiRepoEnv to ensure this is true.
+func enforceSingleFormat(envSet map[string]*DoltEnv) {
```

Comment:
add a bats for this behavior please, someone is definitely going to touch this area again and break something...